### DISCUSSION TITLE
Questions on how to use the digital inputs

### DISCUSSION THREAD

**Discussion by cuishuang413** [phil-barrett/grblHAL-teensy-4.x#70] [General]
Hello Phil,

Thanks for making all of this available for us. I wonder if you can point me to the right direction here. I'm trying to use the digital input pins to detect alarms from my servo drivers and throw an alarm to the Gcode sender. How do I make this happen? I assume I need to do some configuration or coding on the driver? I have been searching for a few days but only got more lost... Thanks a lot.

HC

**Comment by phil-barrett** [comment]
Several ways to do it.  A lot of people run the error input into the halt pin.  Or, you could define a fake extra axis and use the limit pin.

**Reply by cuishuang413** [reply]
Thanks for your reply. Using the halt pin is a good idea. What would you suggest if I want to learn more about using the inputs?

Thanks!

**Comment by phil-barrett** [comment]
You could look at the schematic in the user manual.
